### PR TITLE
cron shall not operate in case we are in maintenance mode

### LINKED ISSUE #14843: cron.php must not run in maintenance mode is enabled
### Steps
1. Setup OC with either ajax cron or CLI cron
2. Switch on maintenance mode
3. `update oc_jobs set last_run=0`
4. Run `php cron.php` or `curl http://userid:password/owncloud/cron.php`
5. Check "oc_jobs" table

### Expected result
Cron does NOT run, timestamps left as is

### Actual result
Cron runs through the background jobs and updates the "last_run" field.

This can have nasty consequences especially when doing manual database repairs, there is a risk that the background file watcher kicks in and does some scanning.

### Versions
Observed in OC 6.0.5, likely to exist in OC 7 and OC 8 too

@DeepDiver1975 @karlitschek @MorrisJobke @icewind1991

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #14843

@PVince81 @karlitschek

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
we shall fix this on all stable branches ... prs are rolling in ...

**Comment by karlitschek** [comment]
not tested but looks good :+1:

File: cron.php [hunk 1]
```diff
@@ -26,7 +26,11 @@
 
 	if (\OCP\Util::needUpgrade()) {
 		\OCP\Util::writeLog('cron', 'Update required, skipping cron', \OCP\Util::DEBUG);
-		exit();
+		exit;
+	}
+	if (\OC::$server->getSystemConfig()->getValue('maintenance', false)) {
+		\OCP\Util::writeLog('cron', 'We are in maintenance mode, skipping cron', \OCP\Util::DEBUG);
```
Comment (nickvergessen) [inline cron.php]: `\OC::$server->getLogger()` :see_no_evil:

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Please backport to stable6, 7 and 8

**Comment by karlitschek** [comment]
backport is approved

**Comment by MorrisJobke** [comment]
stable8 #14845
stable7 #14846

@DeepDiver1975 Can you prepare a stabel6 backport? Thanks

**Pull request merged**